apps/indexer/lib/indexer/fetcher/on_demand/coin_balance.ex

```diff
@@ -34,84 +46,112 @@ defmodule Indexer.Fetcher.OnDemand.CoinBalance do
+    else
+      latest_block_number = latest_block_number()
 
-    case stale_balance_window(latest_block_number) do
-      {:error, _} ->
-        :current
+      case stale_balance_window(latest_block_number) do
+        {:error, _} ->
+          :current
 
-      stale_balance_window ->
-        do_trigger_fetch(address, latest_block_number, stale_balance_window)
+        stale_balance_window ->
+          do_trigger_fetch(address, latest_block_number, stale_balance_window)
+      end
     end
   end
 
   @spec trigger_historic_fetch(Hash.Address.t(), non_neg_integer()) :: balance_status
   def trigger_historic_fetch(address_hash, block_number) do
-    do_trigger_historic_fetch(address_hash, block_number)
+    if __MODULE__.Supervisor.disabled?() do
+      :current
+    else
+      do_trigger_historic_fetch(address_hash, block_number)
+    end
   end
 
-  ## Callbacks
+  @doc false
+  def child_spec([init_options, gen_server_options]) do
+    {state, mergeable_init_options} = Keyword.pop(init_options, :json_rpc_named_arguments)
 
-  def child_spec([json_rpc_named_arguments, server_opts]) do
-    %{
-      id: __MODULE__,
-      start: {__MODULE__, :start_link, [json_rpc_named_arguments, server_opts]},
-      type: :worker
-    }
-  end
+    unless state do
+      raise ArgumentError,
+            ":json_rpc_named_arguments must be provided to `#{__MODULE__}.child_spec " <>
+              "to allow for json_rpc calls when running."
+    end
+
+    merged_init_opts =
+      @defaults
+      |> Keyword.merge(mergeable_init_options)
+      |> Keyword.put(:state, state)
 
-  def start_link(json_rpc_named_arguments, server_opts) do
-    GenServer.start_link(__MODULE__, json_rpc_named_arguments, server_opts)
+    Supervisor.child_spec({BufferedTask, [{__MODULE__, merged_init_opts}, gen_server_options]}, id: __MODULE__)
   end
 
-  @impl true
-  def init(json_rpc_named_arguments) do
-    {:ok, %{json_rpc_named_arguments: json_rpc_named_arguments}}
+  @impl BufferedTask
+  def init(initial, _, _) do
+    initial
   end
 
-  @impl true
-  def handle_cast({:fetch_and_update, block_number, address}, state) do
-    result = fetch_and_update(block_number, address, state.json_rpc_named_arguments)
+  @impl BufferedTask
+  def run(entries, json_rpc_named_arguments) do
+    {fetch_and_update_params, fetch_and_import_params, daily_params} =
+      Enum.reduce(entries, {[], [], []}, fn
+        {:fetch_and_update, block_number, address}, {acc, other_1, other_2} ->
+          {[{block_number, address} | acc], other_1, other_2}
 
-    with {:ok, %{addresses: addresses}} <- result do
-      Accounts.drop(addresses)
-    end
+        {:fetch_and_import, block_number, address}, {other_1, acc, other_2} ->
+          {other_1, [{block_number, address} | acc], other_2}
 
-    {:noreply, state}
-  end
+        {:fetch_and_import_daily_balances, block_number, address}, {other_1, other_2, acc} ->
+          {other_1, other_2, [{block_number, address} | acc]}
+      end)
 
```

Comment:
No call to action, just a way, how to simplify code.
```suggestion
    entries_by_type = Enum.group_by(entries, &elem(&1, 0), &Tuple.delete_at(&1, 0))
  
    {fetch_and_update_params, fetch_and_import_params, daily_params} = {entries_by_type[:fetch_and_update], entries_by_type[:fetch_and_import], entries_by_type[:fetch_and_import_daily_balances]}
```
Last line with assign carrying demonstrative purpose

Reply:
Processed in cb562fe9290cda8c080a729c1f187ddcb6dbc733